test/test_before_build.py

```diff
@@ -14,26 +14,21 @@
 
         # assert that the Python version as written to pythonversion_bb.txt in the CIBW_BEFORE_BUILD step
         # is the same one as is currently running.
-        version_file = 'c:\\pythonversion_bb.txt' if sys.platform == 'win32' else '/tmp/pythonversion_bb.txt'
-        with open(version_file) as f:
+        with open('pythonversion_bb.txt') as f:
             stored_version = f.read()
         print('stored_version', stored_version)
         print('sys.version', sys.version)
         assert stored_version == sys.version
 
-        # check that the executable also was written
-        executable_file = 'c:\\pythonexecutable_bb.txt' if sys.platform == 'win32' else '/tmp/pythonexecutable_bb.txt'
-        with open(executable_file) as f:
-            stored_executable = f.read()
-        print('stored_executable', stored_executable)
-        print('sys.executable', sys.executable)
+        # check that the prefix also was written
+        with open('pythonprefix_bb.txt') as f:
+            stored_prefix = f.read()
+        print('stored_prefix', stored_prefix)
+        print('sys.prefix', sys.prefix)
+        #  Works around path-comparison bugs caused by short-paths on Windows e.g.
+        #  vssadm~1 instead of vssadministrator
 
-        # windows/mac are case insensitive
-        stored_path = os.path.realpath(stored_executable).lower()
-        current_path = os.path.realpath(sys.executable).lower()
-
-        # TODO: This is not valid in an virtual environment
-        assert stored_path == current_path, '{0} != {1}'.format(stored_path, current_path)
+        assert os.stat(stored_prefix) == os.stat(sys.prefix)
```

Comment:
Why not os.path.samefile?

Reply:
There is also at least one more existing os.stat compare that would read better this way, imo. But could be a follow up because this matches the existing one(s).

Reply:
This check was already used in test_before_test.py so I reused it here.
I will change it for samefile which seems better suited.